get_best_selectors

mluscon opened the pull request:
- used nevra_possibilities_real for subject split
- new test for get_best_selectors

jsilhan commented:
There's missing documentation of matching all packages with the same name first from `<package-spec>`: https://github.com/rpm-software-management/dnf/blob/master/doc/command_ref.rst#specifying-packages . It should work the same for `reinstall`, `downgrade`, `upgrade` and `distro-sync`. Please, add the new functionality to them too.

jsilhan commented:
Also change the commit message to `package-spec matches all packages which the name glob pattern fits` or so. It has nothing to do with excludes.

jsilhan commented on dnf/subject.py:
```diff
@@ -148,3 +148,19 @@ def get_best_selector(self, sack, forms=None):
+
+            if nevra and nevra.name and is_glob_pattern(nevra.name):
+                    sltrs = []
+                    pkgs = sack.query().filter(name__glob=nevra.name)
```
Just have seen some scenario in other bug report. There should be `nevra.to_query()` instead. If you had available `pkg A.x86_64, pkg A-lib.i686` then `dnf install A*.x86_64` would fail with error like: `can't install A-lib.x86_64`

jsilhan commented:
@mluscon , stop your hard work on `hawkey.to_nevra()` refactoring, we already have the function for transfering nevra glob attributes to query. It should be `self._nevra_to_filters(sack.query(), nevra)`. No change in hawkey is needed.

mluscon commented:
- changed commit message
- added get_best_selectors into upgrade and distro-sync too
- used _nevra_to_filters() 
- fixed documentation
- changed tests

rholy commented:
Pylint:
```E1103: tests.test_install:209,8: Instance of 'TestSack' has no 'add_excludes' member (but some types could not be inferred)```
Could you add it to our false positives (```scripts/pylint_false_positives```) please?

mluscon commented:
@rholy: done

jsilhan commented:
Merged, thanks
